### DISCUSSION TITLE
Is it possible to use limes XML with an HXP?

### DISCUSSION THREAD

**Discussion by moxie-coder** [openfl/lime#1760] [Q&A] [answered]
I always wondered this since I’m not sure if they’ll work together when compiling with an project.xml, or if it'll just do both

EDIT: I meant to ask this since it seems like it works just fine with hxml files

**Comment by player-03** [comment, marked as answer]
If you supply both project.xml and project.hxp, Lime will pick only one to use (forget which off the top of my head). There are a couple more advanced options available, though. What exactly are you planning to use this for?

**Reply by moxie-coder** [reply]
I saw an project [that seemed to use both](https://github.com/EliteMasterEric/EnigmaEngine), and I thought it’d be more flexible than xml

**Reply by player-03** [reply]
I think they started with project.xml, switched to project.hxp, and forgot to delete the original file.

In any case, I can confirm that the HXP format is more flexible. It can do everything project.xml can do (if in doubt, you can copy the code verbatim from [`ProjectXMLParser`](https://github.com/openfl/lime/blob/develop/src/lime/tools/ProjectXMLParser.hx)). The main downsides are that code completion tends not to work and it isn't as well documented.
